### DISCUSSION TITLE
Is there any equivalent of the Spacer() element from SwiftUI?

### DISCUSSION THREAD

**Discussion by alexmocanu** [AparokshaUI/Adwaita#19] [Q&A] [answered]
Hello, 

Is there anything similar to the Spacer() control from SwiftUI in Adwaita-swift? 

In SwiftUI I would write something like this:
```
HStack {
    Text("Left Text")                    
    Spacer()
    Text("Right Text")    
}
```
The first text would be pushed to the left and the second to the right. 
https://developer.apple.com/documentation/swiftui/spacer

What is the Adwaita equivalent, or is there an alternative to this? 
Sorry if my question seems basic. I'm familiar with SwiftUI, but I'm still trying to find my way around Adwaita.

Here is an example of what I'm trying to do:
```
ListBox(self.items, content: {element in 
	HStack {
		Text(element.title).halign(.start)
		Text(element.numItems).halign(.end)
	}.padding()
})
```
The first text should be aligned to the left and the second to the right, both on the same line. But they're both aligned to the left:
![texts in hstack](https://github.com/AparokshaUI/adwaita-swift/assets/834418/e920384b-bcbb-4b24-a562-1c5719c99057)

If I remove the HStack, the alignment works, but each text will be displayed on a separate line. 
![texts without hstacks](https://github.com/AparokshaUI/adwaita-swift/assets/834418/791d86c1-59cd-4ee1-8768-c4067d34c749)

What I'm looking to do is something like this:
![what_i_want](https://github.com/AparokshaUI/adwaita-swift/assets/834418/3a99af11-bce1-488e-9329-82249edfe09e)

**Comment by alexmocanu** [comment]
In the end, after taking a look at the upstream documentation for  Adwaita, I improvised something using the xalign and frame methods:
```
HStack {
	Text(element.title).xalign(0).frame(minWidth: 130).padding()
	Text(element.numItems).xalign(2).padding()
}
```
![final_result](https://github.com/AparokshaUI/adwaita-swift/assets/834418/89612b31-3c99-4359-852e-49e08ff99cc1)

**Comment by david-swift** [comment, marked as answer]
Have you tried:
```swift
HStack {
    Text("Left Text")
        .hexpand()
        .halign(.start)
    Text("Right Text")
}
```

`halign` makes the widget align inside the widget's frame (the one of `Text` itself, not of `HStack`). Therefore, you need to make the widget's frame expand via `hexpand`. The frame of the parent (`HStack`) will automatically expand as well.

You have to set `halign` to `start` to make sure the widget aligns correctly inside the expanded frame.

Because the width of `Text("Right Text)` is the width of the text itself (no expanding), setting `halign(.end)` is not necessary.

Exactly the same result could be accomplished by expanding the right text:
```swift
HStack {
    Text("Left Text")
    Text("Right Text")
        .hexpand()
        .halign(.end)
}
```

**Reply by alexmocanu** [reply]
Thank You, 

Your suggestions worked and are a lot better than my improvisations. 
I went with the second one (expanding the text on the right).
